# MEG3 Methylation-Seq Demo Project

### Objective
Analyze methylation patterns near the MEG3 locus in human islets from normal vs type II diabetic patients 
by targeted bisulfite sequencing.

Reference article: PMID: 26236400

---

### Dataset
- **Accession:** SRR2052491 & SRR2052492
- **Source:** GEO (GSE69595)
- **Download tool:** SRA Toolkit (`fasterq-dump`)
- **Read Type:** Paired-end, 150 bp

---

### Steps

-> The raw reads were quality checked, filtered, trimmed and adaptors removed as required by fastp.
-> After a second round of quality checking, the reads were aligned to a custom reference genome as mentioned in the article.
-> The alignments and methylation extraction steps were done using Bismark.
-> The reference genome used was hg19. ALthough it was not mentioned specifically in the original article, the coordinates seemed to match when using hg19.
-> Coverage files generated by bismark were converted to csv through awk.
-> The CSV files were used to create plots for relative promoter methylation levels between normal and diabetic samples.
-> The BAM and bedGraph files were used in IGV for visual confirmation of increased methylation in the diabetic sample.
-> The actual CLI scripts are listed below. The python scripts can be found inside the jupyter notebook.

#### Scripts

1. **Make directories**
    ```bash
    mkdir -p "meg3_methylation_demo/"{raw_data,trimmed,fastqc,bismark,results}
    ```

2. **Download data**
    ```bash
    fasterq-dump SRR2052491 --split-files -O raw_data --progress
    ```
    ```bash
    fasterq-dump SRR2052492 --split-files -O raw_data --progress
    ```

3. **Quality control**
    ```bash
    fastqc *.fastq -o ../fastqc/01_qc 
    ```

4. **Trimming**
    ```bash
    for R1 in *_1.fastq; do 
    SAMPLE=$(basename "$R1" _1.fastq)
    R2="${SAMPLE}_2.fastq"
    fastp -i "$R1" -I "$R2" -o "../trimmed/${SAMPLE}_trimmed_1.fastq" -O "../trimmed/${SAMPLE}_trimmed_2.fastq" --trim_poly_x --detect_adapter_for_pe --html "${SAMPLE}_fastp.html" --json "${SAMPLE}_fastp.json" --thread 4
    done
    ```

5. **Prepping the custom fasta reference**
    ```bash
    samtools faidx hg19_chr14/chr14.fa
    samtools faidx hg19_chr14/chr14.fa chr14:101,291,952-101,292,257 > meg3_promoter.fa
    ```

6. **Alignment**
    ```bash

    bismark_genome_preparation --bowtie2 bismark_genome/meg3/

    bismark --bowtie2 bismark_genome/meg3/ -1 ../trimmed/SRR2052491_trimmed_1.fastq -2 ../trimmed/SRR2052491_trimmed_2.fastq -o bismark_alignment_output/

    bismark --bowtie2 bismark_genome/meg3/ -1 ../trimmed/SRR2052492_trimmed_1.fastq -2 ../trimmed/SRR2052492_trimmed_2.fastq -o bismark_alignment_output/
    ```

7. **Methylation Extraction**
    ```bash
    bismark_methylation_extractor --bedGraph --cytosine_report --genome_folder ../../reference/bismark_genome/meg3/ ../bismark_alignment_output/SRR2052491_trimmed_1_bismark_bt2_pe.bam

    bismark_methylation_extractor --bedGraph --cytosine_report --genome_folder ../../reference/bismark_genome/meg3/ ../bismark_alignment_output/SRR2052492_trimmed_1_bismark_bt2_pe.bam
    ```

8. **Experimenting with MethylDackel M-bias**

    **Sorting & Indexing**
    ```bash
    samtools sort -@ 4 -o bismark/bismark_alignment_output/SRR2052491.sorted.bam bismark/bismark_alignment_output/SRR2052491_trimmed_1_bismark_bt2_pe.bam

    samtools index bismark/bismark_alignment_output/SRR2052491.sorted.bam

    samtools sort -@ 4 -o bismark/bismark_alignment_output/sorted/SRR2052492.sorted.bam bismark/bismark_alignment_output/SRR2052492_trimmed_1_bismark_bt2_pe.bam

    samtools index bismark/bismark_alignment_output/sorted/SRR2052492.sorted.bam
    ```
    **M-bias**
    ```bash
    methylDackel mbias reference/meg3_promoter.fa bismark/bismark_alignment_output/sorted/SRR2052491.sorted.bam meth_dackel/mbias_491_

    methylDackel mbias reference/meg3_promoter.fa bismark/bismark_alignment_output/sorted/SRR2052492.sorted.bam meth_dackel/mbias_492_
    ```

9. **Visualisation**
    ```bash
    awk '{print $2","$4","$5","$6}' bismark/bismark_extract/2052491/SRR2052491_trimmed_1_bismark_bt2_pe.bismark.cov > methylation_viz/meg3_methylation_norm.csv

    awk '{print $2","$4","$5","$6}' bismark/bismark_extract/2052492/SRR2052492_trimmed_1_bismark_bt2_pe.bismark.cov > methylation_viz/meg3_methylation_diab.csv

    for file in methylation_viz/*.csv; do
        sed -i '' '1s/^/position,percent_methylation,methylated,unmethylated\n/' $file 
    done
    ```
    ```Python
    # Code listed in Jupyter Notebook.
    ```

---
### Directory Structure

meg3_methylation_demo
├── bismark
│   ├── bismark_alignment_output
│   └── bismark_extract
├── fastqc
│   ├── 01_qc
│   └── trim_qc
├── meth_dackel
├── methylation_viz
├── raw_data
├── reference
│   ├── bismark_genome
│   └── hg19_chr14
├── results
└── trimmed

15 directories

---
### System Info
- macOS Intel, 8 GB RAM
- Conda environments: `rnaseq`, includes `sra-tools`, `fastqc`, `multiqc`, `fastp`, `samtools`, `Bismark`, `MethylDackel`, ``